From 18faa57bc3625e4ed520f852c23b6b71f26410a6 Mon Sep 17 00:00:00 2001
Message-Id: <18faa57bc3625e4ed520f852c23b6b71f26410a6.1432164088.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Thu, 14 May 2015 18:41:46 +0530
Subject: [PATCH] FOR_UPSTREAM [VPG]: drm/i915: Use proper forcewake range for
 blitter ring.

The blitter ring is part of the common well, thus we need to wake up
both render & media before accessing its registers. The bspec currently
has an error where it states that the blitter register range is in the
non forcewake list.

This omission has been passed onto the hardware team.

Issue: GMINL-9414
Change-Id: I2a5fe6e29bc55a3e6730daf5cc682d3587149a97
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_uncore.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_uncore.c b/drivers/gpu/drm/i915/intel_uncore.c
index 21268e2..7d96124 100644
--- a/drivers/gpu/drm/i915/intel_uncore.c
+++ b/drivers/gpu/drm/i915/intel_uncore.c
@@ -550,7 +550,8 @@ void assert_force_wake_inactive(struct drm_i915_private *dev_priv)
 	 REG_RANGE((reg), 0x8000, 0x8300) || \
 	 REG_RANGE((reg), 0x8500, 0x8600) || \
 	 REG_RANGE((reg), 0x9000, 0xB000) || \
-	 REG_RANGE((reg), 0xF000, 0x10000))
+	 REG_RANGE((reg), 0xF000, 0x10000) || \
+	 REG_RANGE((reg), 0x22000, 0x24000))
 
 static void
 ilk_dummy_write(struct drm_i915_private *dev_priv)
-- 
1.7.9.5

